blocks/bricks/sequence_generators.py

```diff
@@ -162,7 +162,7 @@ def cost(self, application_call, outputs, mask=None, **kwargs):
 
         # Separate the deliverables
         states = {name: results[name][:-1] for name in self.state_names}
-        glimpses = {name: results[name] for name in self.glimpse_names}
+        glimpses = {name: results[name][1:] for name in self.glimpse_names}
```

Comment:
I use `return_initial_state=True` here when calling `transition.apply`. That means that I get my initial states and initial glimpses, which are typically all zero. But the glimpses are computed _before_ the word generation. That is 
I need the state number zero and the glimpses number one to compute the probability of the output number one.